rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/CompactView.java

```diff
@@ -166,19 +190,62 @@ private void changeSelection(CompactFlowPanel flow, int id) {
     @Override
     public void updateByDelta(List<NodeSource> nodeSources, List<Node> nodes) {
         /* first call : create the components */
+        if (nodeSourceAdded(nodeSources)) {
+            treeView.sortCompactView(this);
+        }
+        List<NodeSource> currentNodeSources = nodeSourceDeleted(nodeSources) ? nodeSources
+                                                                             : getSortedNodeSourceList(nodeSources);
         if (this.compactPanel == null) {
             initializePanel();
         }
-        updateCompactPanel(nodeSources, nodes);
+        updateCompactPanel(currentNodeSources, nodes);
+        updateMyNodesCompactPanel(currentNodeSources, nodes);
+        updateSelection();
+    }
+
+    private void updateSelection() {
+        if (previousSelectedNs != null) {
+            nodeSourceSelected(previousSelectedNs);
+        } else if (previousSelectedHost != null) {
+            hostSelected(previousSelectedHost);
+        } else if (previousSelectedNode != null) {
+            nodeSelected(previousSelectedNode);
+        }
+    }
 
-        updateMyNodesCompactPanel(nodeSources, nodes);
+    private List<NodeSource> getSortedNodeSourceList(List<NodeSource> nodeSources) {
+        List<NodeSource> sortedNodeSources = new LinkedList<>();
+        ListGridRecord[] sortedRecords = treeView.getTreeGrid().getRecords();
+        if (!nodeSources.isEmpty()) {
+            Arrays.asList(sortedRecords)
+                  .forEach(record -> nodeSources.stream()
+                                                .filter(nodeSource -> nodeSource.getSourceName()
+                                                                                .equals(record.getAttribute(NODE_SOURCES)))
+                                                .findFirst()
+                                                .ifPresent(sortedNodeSources::add));
+        }
+        return sortedNodeSources;
+    }
+
+    private boolean nodeSourceDeleted(List<NodeSource> nodeSources) {
+        return nodeSources == null || nodeSources.isEmpty() || nodeSources.stream().allMatch(NodeSource::isRemoved);
+    }
+
+    private boolean nodeSourceAdded(List<NodeSource> nodeSources) {
+        return nodeSources != null && nodeSources.size() == 1 && nodeSources.get(0).getDeploying().size() != 0;
     }
 
     private void updateCompactPanel(List<NodeSource> nodeSources, List<Node> nodes) {
         processNodeSources(compactPanel, nodeSources);
         processNodes(compactPanel, nodes);
     }
 
+    public void removeNodeSources(List<NodeSource> currentNodeSources) {
+        if (currentNodeSources != null) {
```

Comment:
if (currentNodeSources) {

Reply:
currentNodeSources it's a list, not a boolean so it will not work like this

Reply:
it should work for list as well